src/client/datascience/errors/errorHandler.ts

```diff
@@ -104,23 +102,11 @@ export class DataScienceErrorHandler implements IDataScienceErrorHandler {
         if (err instanceof JupyterKernelDependencyError) {
             return err.reason;
             // Use the kernel dependency service to first determine if this is because dependencies are missing or not
-        } else if (
-            err instanceof JupyterInstallError ||
-            ((purpose === 'start' || purpose === 'restart') &&
-                !(await this.kernelDependency.areDependenciesInstalled(kernelConnection, undefined, true)))
-        ) {
-            const tokenSource = new CancellationTokenSource();
-            try {
-                return this.kernelDependency.installMissingDependencies(
-                    resource,
-                    kernelConnection,
-                    new DisplayOptions(false),
-                    tokenSource.token,
-                    true
-                );
-            } finally {
-                tokenSource.dispose();
-            }
+        } else if (err instanceof JupyterInstallError && (purpose === 'start' || purpose === 'restart')) {
```

Comment:
Maybe I'm missing something elsewhere, but this will not work for raw. JupyterInstallError is only thrown in the jupyter cases as far as I remember.

Reply:
I don't see the problem with that, 
If we're dealing with raw, then this error will not be throw, meaning its not a problem

Reply:
No it means we won't install dependencies into a raw kernel. It will just fail.

Reply:
That's why it was an 'or' before. It should install dependencies if it's an install error or the dependency analyzer says it needs the dependencies.

Reply:
> No it means we won't install dependencies into a raw kernel. It will just fail.

Aaah, then we need to throw KernelDependency error, i'll check.
The only way we can throw `JupyterInstallError ` is in the case of non-raw, possible we're missing another condition. I'll check

Reply:
The code used to check before running a raw kernel but I removed that and had it check here after it failed.

Reply:
Got it, fixed,